model/coverage.py

```diff
@@ -242,6 +242,13 @@ def update(self, start=None, finish=None, achievements=None,
         # .exception
         self.exception = exception
 
+    def to_data(self):
+        """Convert this Timestamp to an unfinalized TimestampData."""
```

Comment:
Unfinalized means that it hasn't finished running so what values should it not have until it is finalized?

Reply:
'Unfinalized' means it's missing the information about which service it's a timestamp for. In this case we're deliberately leaving that data out, and that's because I want to force `finalize` to be called on this timestamp before it's written to the database. And _that's_ because `finalize` does other stuff like fill in defaults for any missing values.